scipy/linalg/tests/test_lapack.py

```diff
@@ -1654,3 +1655,80 @@ def test_geqrfp_lwork(dtype, shape):
     m, n = shape
     lwork, info = geqrfp_lwork(m=m, n=n)
     assert_equal(info, 0)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if type(shape) != tuple:
+        # in case shape is not passed in as a tuple
```

Comment:
Got it. I added this so it wouldn’t break gttrf/s since that will need to be updated for how it calls this (it only requests vectors by passing a single n)

Reply:
Ok, make that pass in (n, n)